File: WorkManagerJava/app/src/main/java/com/websarva/wings/android/workmanagerjava/CountWithIdWorker.java
package com.websarva.wings.android.workmanagerjava;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Data;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.util.UUID;

public class CountWithIdWorker extends Worker {
	public CountWithIdWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
		super(context, workerParams);
	}

	@NonNull
	@Override
	public Result doWork() {
		UUID uuid = getId();
		Log.i("CountWithIdWorker", "IDが" + uuid + "で5回ループを行います。");
		Data.Builder dataBuilder = new Data.Builder();
		dataBuilder.putString("id", uuid.toString());
		Data outputData = dataBuilder.build();
		Result returnVal = Result.success(outputData);
		for(int i = 1; i <= 5; i++) {
			Log.i("CountWithIdWorker", "IDが" + uuid + "の" +  i + "回目");
			try {
				Thread.sleep(1000);
			} catch (InterruptedException e) {
				returnVal = Result.failure();
			}
		}
		return returnVal;
	}
}

File: WorkManagerKotlin/app/src/main/java/com/websarva/wings/android/workmanagerkotlin/CountWithIdWorker.kt
package com.websarva.wings.android.workmanagerkotlin

import android.content.Context
import android.util.Log
import androidx.work.Data
import androidx.work.Worker
import androidx.work.WorkerParameters

class CountWithIdWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
	override fun doWork(): Result {
		Log.i("CountWorker", "IDが${id}で5回ループを行います。")
		val dataBuilder = Data.Builder()
		dataBuilder.putString("id", id.toString())
		val outputData = dataBuilder.build()
		for(i in 1..5) {
			Log.i("CountWorker", "IDが${id}の${i}回目")
			Thread.sleep(1000)
		}
		return Result.success(outputData)
	}
}
